fbw-a380x/src/systems/instruments/src/Common/EWDMessageParser.tsx

```diff
@@ -0,0 +1,169 @@
+                            className={`Underline ${color}Line`}
+                            d={`M ${x + (startCol * LETTER_WIDTH)}
+                            ${y + (lines.length * LINE_SPACING) - 22} h ${(col - startCol) * LETTER_WIDTH + 12} v 27 h ${-((col - startCol) * LETTER_WIDTH + 12)} v -27`}
+                            strokeLinecap="round"
+                        />,
+                    );
+                }
+
+                startCol = col;
+            }
+
+            if (char === '\x1B') {
+                let ctrlBuffer = '';
+                i++;
+                for (; i < message.length; i++) {
+                    ctrlBuffer += message[i];
+
+                    let match = true;
+                    switch (ctrlBuffer) {
+                    case 'm':
+                        // Reset attribute
+                        underlined = false;
+                        // flashing = false;
+                        framed = false;
+                        break;
+                    case '4m':
+                        // Underlined attribute
+                        underlined = true;
+                        break;
+                    case ')m':
+                        // Flashing attribute
+                        // flashing = true;
+                        break;
+                    case "'m":
+                        // Characters which follow must be framed
+                        framed = true;
+                        break;
+                    case '<1m':
+                        // Select YELLOW
+                        color = 'Yellow';
+                        break;
+                    case '<2m':
+                        // Select RED
+                        color = 'Red';
+                        break;
+                    case '<3m':
+                        // Select GREEN
+                        color = 'Green';
+                        break;
+                    case '<4m':
+                        // Select AMBER
+                        color = 'Amber';
+                        break;
+                    case '<5m':
+                        // Select CYAN (blue-green)
+                        color = 'Cyan';
+                        break;
+                    case '<6m':
+                        // Select MAGENTA
+                        color = 'Magenta';
+                        break;
+                    case '<7m':
+                        // Select WHITE
+                        color = 'White';
+                        break;
+                    default:
+                        match = false;
+                        break;
+                    }
+
+                    if (match) {
+                        break;
+                    }
+                }
+
+                continue;
+            }
+
+            if (char === '\r') {
+                lines.push(<text x={x} y={y + (lines.length * LINE_SPACING)}>{spans}</text>);
+
+                spans = [];
```

Comment:
```suggestion
                spans.length = 0;
```

Optimization for reducing garbage collection.

Reply:
EWDMessageParser wasn't changed by me, and removed via the latest commit (not used for build right now, might be better to refactor to fbw_common).